vite.config.js

```diff
@@ -0,0 +1,56 @@
+import react from "@vitejs/plugin-react-swc";
+import { defineConfig, transformWithEsbuild } from "vite";
+import cssInjectedByJsPlugin from "vite-plugin-css-injected-by-js";
+import entryPoints from "./webpack.config.entry";
+
+/**
+ * Vite is opinionated and doesn't support JSX in .js files (and that's cool).
+ * This plugin makes it clear that we don't care about its opinions: process
+ * all .js files with esbuild as if they were .jsx.
+ *
+ * This has a small performance penalty compared to renaming the files, but
```

Comment:
I imagine the performance penalty isn't that bad when you consider the built-in IDE functionality we get when `.jsx` or `.tsx` files.